chat-plugins/mafia.js

```diff
@@ -72,6 +74,97 @@ class MafiaPlayer extends Rooms.RoomGamePlayer {
 	}
 }
 
+// Parses a single role into an object
+function parseRole(roleString) {
+	let target = roleString.toLowerCase().replace(/[^\w\d\s]/g, '').split(' ');
+	let role = {
+		name: roleString.split(' ').map(p => { return toId(p) === 'solo' ? '' : p; }).join(' '),
+		memo: ['During the Day, you may vote for whomever you want lynched.'],
+		alignment: '',
+		problems: [],
+	};
+	role.safeName = Chat.escapeHTML(role.name);
+	role.id = toId(role.name);
+	for (let key in MafiaData.roles) {
+		if (key.includes('_')) {
+			let roleKey = target.slice().map(p => { return toId(p); }).join('_');
```

Comment:
This could just be `target.slice().map(p => toId(p)).join('_');` for brevity since arrow functions implicitly return.